index.rst

```diff
@@ -1,29 +1,375 @@
+However, it allows easy deployment of basic quota support to every service.
+More sophisticated quotas can be added later if needed.
+
+Services that need to apply quotas directly can request a delegated token (possibly with no scopes if they don't need a delegated token for anything else) and then use that token to retrieve quota information via ``/auth/api/v1/user-info``.
+
+Quota settings
+==============
+
+For the initial implementation, we will add quota information to the Gafaelfawr configuration via the Helm chart.
+
+Quota configuration
+-------------------
+
+The data structure for the quota information will look something like this:
+
+.. code-block:: yaml
+
+   quotas:
+     default:
+       notebook:
+         cpu: 9
+         memory: "27Gi"
+       api:
+         datalinker: 500
+         hips: 2000
+         tap: 500
+         vo-cutouts: 100
+     groups:
+       g_developers:
+         api:
+           datalinker: 500
+
+The ``default`` key establishes default quotas for every user.
+The ``groups`` key provides additional quotas to particular groups.
+These quotas are additive, so in the above case a user who is a member of the ``g_developers`` group would have a quota of 1000 for the ``datalinker`` service.
+
+API quotas are in requests per fifteen minutes.
+This is an awkward interval, but (as discussed in :ref:`rate-limiting`) the interval is also the length of time that the user will be blocked from accessing the service.
+One minute seems to short, and one hour (used by GitHub) seems too long.
+
+The keys under ``api`` are the names of the services, as configured in the Gafaelfawr auth URL for that service.
+Normally, this is set in the ``config`` section of the corresponding ``GafaelfawrIngress`` custom resource.
+
+A given API service does not have to have a quota.
+If no quota is configured, the quota for all users is unlimited and requests won't be tracked.
+
+Quota overrides
+---------------
+
+Emergency override information will be stored in the Gafaelfawr Redis under the key ``quota-override``.
+The value of the key will be a JSON document such as the following:
+
+.. code-block:: json
+
+   {
+       "default": {
+           "notebook": {
+               "spawn": false,
+               "cpu": 4
+           },
+           "api": {
+               "datalinker": 10
+           }
+       },
+       "groups": {
+           "g_users": {
+               "api": {
+                   "vo-cutouts": 10
+               }
+           }
+       },
+       "bypass": [
+           "g_admins"
+       ]
+   }
+
+This mostly has the same structure as the configuration, but it overrides all quota information taken from the configuration, including additions from groups.
+So, for example, if the above override were in place, all users would have a quota of 10 for the datalinker API, including members of ``g_developers`` who normally get an extra 500.
```

Comment:
I understand, particularly given the awkwardness of the interval, why you might not want to, but something in my brain is shrieking at the dimensionless units here and would really like to see "queries per quarter-hour" or something after "10".